kafka-client/src/main/java/dev/responsive/kafka/internal/db/LwtWriterFactory.java

```diff
@@ -85,6 +85,12 @@ public RemoteWriteResult<P> setOffset(final long offset) {
     builder.addStatement(fencingStatement(tablePartition));
     builder.addStatement(table.setOffset(kafkaPartition, offset));
 
+    // TODO(sophie): clean up this hack, perhaps by combining the offset and stream-time into
+    //  a single metadata row update
+    if (table instanceof CassandraWindowedTable) {
+      builder.addStatement(((CassandraWindowedTable) table).setStreamTime(kafkaPartition, epoch));
```

Comment:
since this is being issued as part of the batch, we probably don't need the:
```
ifColumn(EPOCH.column()).isEqualTo(bindMarker(EPOCH.bind()))
```
part of the query. the fencing statement will do that for us (i'm not sure if there's any problem with checking that twice, probably not)

Reply:
ah, right, good point. I'll just remove it for now

But actually this is something I was thinking about the other day -- my understanding is that functionally, there is no difference between doing the fencing with a separate epoch-checking statement vs adding an epoch-check to each individual insert/delete. But is there a performance difference? 

I'm assuming yes, these conditions have a cost, so it's best to just do the epoch-read once per batch write instead of per-update? I assume there's no way Cassandra or Scylla is smart enough to optimize the extra epoch checks away..